tools/data-handler/src/validate.ts

```diff
@@ -423,35 +428,54 @@ export class Validate {
     project: Project,
     card: card,
   ): Promise<string> {
-    let validationErrors = '';
+    const validationErrors: string[] = [];
 
     if (!card.metadata) {
-      validationErrors += `Card '${card.key}' has no metadata. Card object needs to be instantiated with '{metadata: true}'`;
+      validationErrors.push(
+        `Card '${card.key}' has no metadata. Card object needs to be instantiated with '{metadata: true}'`,
+      );
     }
 
     const cardType = await project.cardType(card.metadata?.cardtype);
     if (!cardType) {
-      validationErrors += `Card '${card.key}' has invalid cardtype '${card.metadata?.cardtype}'`;
+      validationErrors.push(
+        `Card '${card.key}' has invalid cardtype '${card.metadata?.cardtype}'`,
+      );
     }
 
     if (cardType) {
       if (!cardType.workflow) {
-        validationErrors += `Cardtype '${cardType?.name}' does not have 'workflow'`;
+        validationErrors.push(
+          `Cardtype '${cardType?.name}' does not have 'workflow'`,
+        );
       }
 
       const workflow = await project.workflow(cardType?.workflow);
       if (workflow) {
-        const states = workflow.states;
-        const cardState = card.metadata?.workflowState;
-        const found = states.find((item) => item.name === cardState);
-        if (!found) {
-          validationErrors += `Card '${card.key}' has invalid state '${cardState}'`;
+        if (!Project.isTemplateCard(card)) {
+          const states = workflow.states;
+          const cardState = card.metadata?.workflowState;
+          const found = states.find((item) => item.name === cardState);
+          if (!found) {
+            validationErrors.push(
+              `Card '${card.key}' has invalid state '${cardState}'`,
+            );
+          }
+        } else {
+          const cardState = card.metadata?.workflowState;
+          if (cardState) {
+            validationErrors.push(
+              `Template card ${card.key} must have empty "workflowState"`,
+            );
+          }
         }
       } else {
-        validationErrors += `Workflow of '${cardType.workflow}' cardtype '${cardType?.name}' does not exist in the project`;
+        validationErrors.push(
+          `Workflow of '${cardType.workflow}' cardtype '${cardType?.name}' does not exist in the project`,
+        );
       }
     }
-    return validationErrors;
+    return validationErrors.join('\n');
```

Comment:
I think it would be better to represent validation errors using a list of either objects or just strings. But that deserves an own PR

Reply:
Looks like #130 does that :smile: